api/src/extensions/manager.ts

```diff
@@ -794,4 +792,20 @@ export class ExtensionManager {
 
 		await Promise.all(unregisterFunctions.map((fn) => fn()));
 	}
+
+	/**
+	 * If extensions must load successfully, any errors will cause the process to exit.
+	 * Otherwise, the error will only be logged as a warning.
+	 */
+	private handleExtensionError(type: ExtensionType, name: string, error: any): void {
+		if (toBoolean(env['EXTENSIONS_MUST_LOAD'])) {
```

Comment:
I prefer casting in `env.ts` like
https://github.com/directus/directus/blob/2e01a75322a13a290587ce9ba9e1bb822ed87742/api/src/env.ts#L362

Reply:
IIRC recently there's been quite a number of places were we favored casting only when we're accessing the values, in places such as:

- https://github.com/directus/directus/blob/2e01a75322a13a290587ce9ba9e1bb822ed87742/api/src/server.ts#L93
- https://github.com/directus/directus/blob/2e01a75322a13a290587ce9ba9e1bb822ed87742/api/src/utils/sanitize-query.ts#L13

along with a past discussion over here on expecting environment variables to be string all the time: https://github.com/directus/directus/pull/17309#discussion_r1091971167, so I was think this is the way going forward. But it can definitely be a separate discussion (maybe we pivoted from that train of thought since then for example) 👍